src/lib/input/input.ts

```diff
@@ -90,14 +95,46 @@ export class MdHint {
   @Input() align: 'start' | 'end' = 'start';
 }
 
+/**
+ * Enables auto expanding behaviour for the textarea
+ */
+@Directive({
+  selector: 'textarea[mdAutosize]'
+})
+export class MdAutosize implements OnInit {
+  @HostListener('input', ['$event.target'])
+  public onChange() {
+    if (this._isActive) {
+      this._adjust();
+    }
+  }
+
+  @Input('mdAutosize')
+  private _isActive: boolean;
+
+  constructor(private _elRef: ElementRef) {}
+
+  ngOnInit() {
+    if (this._isActive) {
+      this._adjust();
+    }
+  }
+
+  private _adjust() {
+    this._elRef.nativeElement.style.overflow = 'hidden';
+    this._elRef.nativeElement.style.height = 'auto';
+    this._elRef.nativeElement.style.height = `${this._elRef.nativeElement.scrollHeight}px`;
```

Comment:
Well.. it's considerably more complex when you have to  deal with the `rows` attribute. So there are two options, not allowing autosize and rows on the same element(because honestly, who's using rows nowadays, it's only good for devices not supporting css) or go with the 10 times more complex code. I started working on the latter, but I'd be much happier with the former.